### PR TITLE
Fixed Markdown In Bangla Books List

### PR DESCRIPTION
**Pull request by HridoyHazard**
## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by davorpa** [comment]
Thanks for helping to address with #4579

File: books/free-programming-books-bn.md [hunk 1]
```diff
@@ -53,7 +53,7 @@
 * [Pattern Printing in C](https://www.youtube.com/playlist?list=PLgH5QX0i9K3oTxQhx2kejYmQn6qtRULCD) - Anisul Islam
 
 
-<h3 id="cpp">C++</h3>
+### C++
```
Comment (davorpa) [inline books/free-programming-books-bn.md:56]: See how it's addresed in other files to be synced with index entry: https://github.com/EbookFoundation/free-programming-books/pull/6621/files#R8

```markdown
* <a href="#cpp">C++</a>
```

Headings with only one letter and symbols makes some problem with rendering bugs across GitHub auto ToC, GMF markdown preview and the HTML rendered by GitHub pages.

```markdown
### <a name="cpp"></a>C++
```

**Comment by HridoyHazard** [comment]
check now fixed according to your idea @davorpa

File: books/free-programming-books-bn.md (hunk 1 above)
Comment (davorpa) [reply books/free-programming-books-bn.md]: Now link doesn't work.

Needs change `href="cpp"` too to have a working Index-To-Section linking.

Check it clicking in `C++` Index item.

**Review by davorpa — CHANGES_REQUESTED** [review]
Not resolved yet

File: books/free-programming-books-bn.md [hunk 2]
```diff
@@ -53,7 +53,7 @@
 * [Pattern Printing in C](https://www.youtube.com/playlist?list=PLgH5QX0i9K3oTxQhx2kejYmQn6qtRULCD) - Anisul Islam
 
 
-### <a name="cpp"></a>C++
+### <a href="#cpp"></a>C++
```
Comment (davorpa) [inline books/free-programming-books-bn.md:56]: Sorry if I explain wrong. It was well with `name`

```suggestion
### <a name="#cpp"></a>C++
```

File: books/free-programming-books-bn.md [hunk 3]
```diff
@@ -53,7 +53,7 @@
 * [Pattern Printing in C](https://www.youtube.com/playlist?list=PLgH5QX0i9K3oTxQhx2kejYmQn6qtRULCD) - Anisul Islam
 
 
-<h3 id="cpp">C++</h3>
+### <a name="#cpp"></a>C++
```
Comment (davorpa) [inline books/free-programming-books-bn.md:56]: Sorry, I fail in suggestion :worried:. Anchors haven't hash

```suggestion
### <a name="cpp"></a>C++
```

**Review by davorpa — APPROVED** [review]

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**